contracts/dao/pcsp/PCSPReward.sol

```diff
@@ -0,0 +1,157 @@
+        transferOwnership(owner);
+    }
+
+    function setPCSPConfiguration(
+        address addressOfPCSPConfiguration
+    )
+        external
+        onlyOwner
+        validPCSPConfiguration(addressOfPCSPConfiguration)
+    {
+        address _oldPCSPConfiguration = _addressOfPCSPConfiguration;
+        _addressOfPCSPConfiguration = addressOfPCSPConfiguration;
+
+        emit SetPCSPConfiguration(
+            _oldPCSPConfiguration, addressOfPCSPConfiguration
+        );
+    }
+
+    function getPCSPConfiguration() external override view returns (address) {
+        return _addressOfPCSPConfiguration;
+    }
+
+    function getRiskOfGettingStroke(
+        uint256 geneNFTTokenID
+    )
+        external override view returns (uint256)
+    {
+        return _riskOfGettingStrokeRecords[geneNFTTokenID];
+    }
+
+    function checkGeneNFTRewardStatus(
+        uint256 geneNFTTokenID
+    )
+        external override view returns (bool)
+    {
+        return _geneNFTRewardStatuses[geneNFTTokenID];
+    }
+
+    function calculateRewardForMultipleCustomers(
+        uint256[] memory geneNFTTokenIDs,
+        uint256[] memory risksOfGettingStroke,
+        uint256[] memory revenuesInPCSP
+    )
+        external
+        override
+        onlyOwner
+    {
+        require(
+            geneNFTTokenIDs.length == risksOfGettingStroke.length,
+            "PCSPReward: list of GeneNFTTokenIds and risksOfGettingStroke must have same length"
+        );
+        require(
+            risksOfGettingStroke.length == revenuesInPCSP.length,
+            "PCSPReward: list of risksOfGettingStroke and revenuesInPCSP must have same length"
+        );
+
+        for(uint256 i = 0; i < geneNFTTokenIDs.length; i++) {
+            _calculateCustomerReward(
+                geneNFTTokenIDs[i], risksOfGettingStroke[i], revenuesInPCSP[i]
+            );
+        }
+
+    }
+
+    function _calculateCustomerReward(
+        uint256 geneNFTTokenID,
+        uint256 riskOfGettingStroke,
+        uint256 revenueInPCSP
+    )
+        private
+    {
+        require(
+            !_geneNFTRewardStatuses[geneNFTTokenID],
+            "PCSPReward: the GeneNFT has rewarded for risk of getting stroke"
+        );
+
+        IPCSPConfiguration config = IPCSPConfiguration(
+            _addressOfPCSPConfiguration
+        );
+        require(
+            config.checkActiveRiskOfGettingStroke(riskOfGettingStroke),
+            "PCSPReward: risk of getting stroke value is invalid"
+        );
+
+        GNFTToken geneNFTToken = GNFTToken(config.getGeneNFTAddress());
+        // retrieve owner of GeneNFT
+        address geneNFTOwner = geneNFTToken.ownerOf(geneNFTTokenID);
+
+
+        // mark GeneNFT rewarded
+        _geneNFTRewardStatuses[geneNFTTokenID] = true;
+        // record risk of getting stroke of customer
+        _riskOfGettingStrokeRecords[geneNFTTokenID] = riskOfGettingStroke;
+
+        emit RecordRiskOfGettingStroke(geneNFTTokenID, riskOfGettingStroke);
+
```

Comment:
có thể xem xét tách ra làm hàm record risk of stroke riêng. Hàm `_calculateCustomerReward` có thể đổi thành `_rewardCustomer` để có thể hiểu thành việc reward cho customer gồm bước là record lại và tính toán lượng reward.